### PR TITLE
3.4 update spatial docs

### PR DESCRIPTION
**Pull request by craigtaverner**
Include examples of min:max range queries, bounding box queries and information about comparability and orderability.

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 1]
```diff
@@ -133,6 +134,26 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
     )
   }
 
+  @Test def use_index_with_where_using_multiple_range_comparisons() {
+    // Need to make index preferable in terms of cost
+    executePreparationQueries((0 to 300).map { i =>
+      "CREATE (:Person)"
+    }.toList)
+    profileQuery(
+      title = "Use index with `WHERE` using multiple range comparisons",
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala]: Reword title as per https://github.com/neo4j/neo4j-documentation/pull/241 for consistency

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 2]
```diff
@@ -133,6 +134,26 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
     )
   }
 
+  @Test def use_index_with_where_using_multiple_range_comparisons() {
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala:136]: Nice addition: this complements the other PR (as that one didn't have multiple range comparisons). Nothing to do here: just a comment.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 3]
```diff
@@ -133,6 +134,26 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
...
+    // Need to make index preferable in terms of cost
+    executePreparationQueries((0 to 300).map { i =>
+      "CREATE (:Person)"
+    }.toList)
+    profileQuery(
+      title = "Use index with `WHERE` using multiple range comparisons",
+      text = "When the `WHERE` clause contains multiple inequality (range) comparisons for the same property these can be combined " +
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala]: "property these can" -> "property, these can"

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 4]
```diff
@@ -133,6 +134,26 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
...
+        "in a single index range seek. " +
+        "If you want Cypher to use specific indexes, you can enforce it using hints. See <<query-using>>.",
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala]: Please remove this line. As per https://github.com/neo4j/neo4j-documentation/pull/241, this general note has been added as a "note", negating the need to pepper this sentence all over the show.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala [hunk 5]
```diff
@@ -217,9 +238,29 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
...
-    assert(expectedIndexes === db.indexPropsForLabel(label) || expectedIndexesAlt === db.indexPropsForLabel(label))
+  @Test def use_index_with_bbox_query() {
+    executePreparationQueries(
+      (for(x <- -10 to 10; y <- -10 to 10) yield s"CREATE (:Person {location: point({x:$x, y:$y})})").toList ++ List(
+        "MATCH (n:Person {firstname: 'Andres'}) SET n.location = point({x: 1.2345, y: 5.4321})",
+        "MATCH (n:Person {firstname: 'Mark'}) SET n.location = point({y: 1.2345, x: 5.4321})"
+      )
+    )
+    profileQuery(
+      title = "Use index when executing a spatial bounding box search",
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala]: Harmonize this title as per https://github.com/neo4j/neo4j-documentation/pull/241

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala [hunk 6]
```diff
@@ -257,9 +258,63 @@ class SpatialTest extends DocumentingTest {
           |<<cypher-spatial-crs, four coordinate reference systems>>.
           |This allows for both <<schema-index-use-a-single-property-index-with-where-using-equality, equality>>
           |and <<schema-index-use-index-with-where-using-range-comparisons, range>> queries using exactly the same syntax and behaviour as for other property types.
+          |If two range predicates are used, setting minimum and maximum points, this will effectively result in a
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: Reword - this is clunky. Maybe "If two range predicates are used **by** setting minimum and maximum points," or similar

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala [hunk 7]
```diff
@@ -257,9 +258,63 @@ class SpatialTest extends DocumentingTest {
...
+          |<<schema-index-use-index-when-executing-a-spatial-bounding-box-search, 'bounding box query'>>.
           |In addition, queries using the `distance` function can, under the right conditions, also use the index, as described in the section
-          |<<schema-index-use-index-when-executing-a-spatial-distance-search, 'Use index when executing a spatial distance search'>>
+          |<<schema-index-use-index-when-executing-a-spatial-distance-search, 'Use index when executing a spatial distance search'>>.
         """.stripMargin)
     }
+    section("Comparability and Orderability", "cypher-comparability-orderability") {
+      p(
+        """
+          |Points with different CRS are not comparable. This means that any function operating on two points of different type will return null.
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: Stylistic issue: please put sentences on separate lines

Comment (petraselmer) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: Format null as elsewhere in docs: `null`

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala [hunk 8]
```diff
@@ -257,9 +258,63 @@ class SpatialTest extends DocumentingTest {
...
+          |This is true of the distance function as well as inequality comparisons. If these are used in predicate, they will cause the associate match
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: More line splitting

Comment (petraselmer) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: I would hyperlink to "the distance function"

Comment (petraselmer) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: "If these are used in predicate" -> ??? "If these are used as predicates" (or did you mean something else?)
"they will cause the associate match" -> "they will cause the associated match"

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala [hunk 9]
```diff
@@ -257,9 +258,63 @@ class SpatialTest extends DocumentingTest {
...
+          |to return no results.
+        """.stripMargin
+      )
+      query(
+        """WITH point({x:3, y:0}) AS p2d, point({x:0, y:4, z:1}) as p3d
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: as -> AS 
(Cypher style guide - please check elsewhere in this PR)

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala [hunk 10]
```diff
@@ -257,9 +258,63 @@ class SpatialTest extends DocumentingTest {
...
+          }
+          withClue("Expect the negative equality test to return true") {
+            (record("p2d <> p3d") == true) should be(true)
+          }
+        })) {
+        resultTable()
+      }
+      p(
+        """
+          |However, all types are orderable. The Point types will be ordered after Numbers and before Temporal types.
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: New sentence, new line

Comment (petraselmer) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: Hyperlink "Point"

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala [hunk 11]
```diff
@@ -257,9 +258,63 @@ class SpatialTest extends DocumentingTest {
...
+          |Points with different CRS with be ordered by their SRID numbers. For the current set of 4 CRS, this means
```
Comment (petraselmer) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: New sentence, new line

Comment (petraselmer) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: 4 -> four

(for small numbers, we write out the number in words)

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala (hunk 7 above)
Comment (petraselmer) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: "two points of different type" -> "two points of different types"

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala (hunk 11 above)
Comment (petraselmer) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: I suggest hyperlinking "CRS" to the text introducing these

**Review by petraselmer — CHANGES_REQUESTED** [review]

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala (hunk 6 above)
Comment (craigtaverner) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: I think you got that back to front. The predicates result in min,max, not the other way round.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala (hunk 10 above)
Comment (craigtaverner) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SpatialTest.scala]: I could not find something to link it to. It is a type, and the section on types simply lists Point along with all other types. Or perhaps you want a link to the `point` function? In that case should the section on types also link to the `point` function?

**Review by petraselmer — APPROVED** [review]

**Comment by petraselmer** [comment]
Nice! 👍

**Review by jjaderberg — CHANGES_REQUESTED** [review]
The following IDs are removed, but still referenced from elsewhere:

- `schema-index-use-index-with-where-using-range-comparisons`
- `schema-index-use-index-when-executing-a-spatial-bounding-box-search`
- `schema-index-use-index-when-executing-a-spatial-distance-search`

**Review by jjaderberg — APPROVED** [review]
:+1:

**Pull request merged**